### PR TITLE
Add unit tests for sha256 functions

### PR DESCRIPTION
**Pull request by hikinggrass**
## Describe your changes
Unit tests for the sha256 functions as a prerequisite for https://github.com/EVerest/libocpp/pull/535

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: tests/v201_utils_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    ASSERT_EQ("e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+              ocpp::v201::utils::sha256(empty_input));
+    ASSERT_EQ("12998c017066eb0d2a70b94e6ed3192985855ce390f321bbdb832022888bd251",
+              ocpp::v201::utils::sha256(short_input));
+    ASSERT_EQ("34aa8868354dc8eb0e76fbc8b5f13259094bcdf5688c6a48a2bcb89ed863d441",
+              ocpp::v201::utils::sha256(long_input));
+}
+
+TEST_F(V201UtilsTest, test_invalid_sha256) {
+    ASSERT_NE(invalid, ocpp::v201::utils::sha256(empty_input));
```
Comment (marcemmers) [inline tests/v201_utils_tests.cpp:38]: I am not so sure what the purpose is of these `NE` tests, especially if you already tested the valid outputs.

File: tests/v201_utils_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    ASSERT_NE(invalid, ocpp::v201::utils::sha256(short_input));
+    ASSERT_NE(invalid, ocpp::v201::utils::sha256(long_input));
+}
+
+TEST_F(V201UtilsTest, test_valid_generate_token_hash) {
+    ocpp::v201::IdToken valid_iso14443_token = {"ABAD1DEA", ocpp::v201::IdTokenEnum::ISO14443};
```
Comment (marcemmers) [inline tests/v201_utils_tests.cpp:44]: Might be interesting to validate that the same token with a different token type results in a different output?

**Review by marcemmers — CHANGES_REQUESTED** [review]

File: tests/v201_utils_tests.cpp (hunk 1 above)
Comment (hikinggrass) [reply tests/v201_utils_tests.cpp]: You're right, I've removed them

File: tests/v201_utils_tests.cpp (hunk 2 above)
Comment (marcemmers) [reply tests/v201_utils_tests.cpp:37]: But now you still have one token with id `valid` and the other `ABAD1DEA` right?

Comment (hikinggrass) [reply tests/v201_utils_tests.cpp:37]: You're right, I added another test that checks for that :+1:

**Review by marcemmers — APPROVED** [review]

**Pull request merged**